src/execution/__tests__/executor-test.js

```diff
@@ -306,11 +307,65 @@ describe('Execute: Handles basic execution tasks', () => {
     const field = operation.selectionSet.selections[0];
     expect(resolvedInfo).to.deep.include({
       fieldNodes: [field],
-      path: { prev: undefined, key: 'result' },
+      path: { prev: undefined, key: 'result', parentType: 'Test' },
       variableValues: { var: 'abc' },
     });
   });
 
+  it('populates path correctly with complex types', () => {
+    let path;
+    const a = new GraphQLObjectType({
+      name: 'A',
+      fields: {
+        test: {
+          type: GraphQLString,
+          resolve(_val, _args, _ctx, info) {
+            path = info.path;
+          },
+        },
+      },
+    });
+    const testUnion = new GraphQLUnionType({
+      name: 'TestUnion',
+      types: [a],
+      resolveType() {
+        return a;
+      },
+    });
+    const testType = new GraphQLObjectType({
+      name: 'Test',
+      fields: {
+        test: {
+          type: new GraphQLNonNull(
+            new GraphQLList(new GraphQLNonNull(testUnion)),
+          ),
+          resolve(_val, _args, _ctx, _info) {
+            return [{ type: 'A' }];
+          },
+        },
+      },
+    });
+    const schema = new GraphQLSchema({ query: testType });
+
+    const document = parse('query { l1: test { ... on A { l2: test } } }');
```

Comment:
this is pretty complex query so it deserves to be written as multi-line similar to some of the tests below in this file.